src/storage/src/region/tests/truncate.rs

```diff
@@ -0,0 +1,240 @@
+    }
+
+    async fn truncate(&self) {
+        self.base().region.truncate().await.unwrap();
+    }
+
+    async fn reopen(&mut self) {
+        // Close the old region.
+        if let Some(base) = self.base.as_ref() {
+            base.close().await;
+        }
+        self.base = None;
+        // Reopen the region.
+        let store_config = config_util::new_store_config(
+            REGION_NAME,
+            &self.store_dir,
+            EngineConfig {
+                max_files_in_l0: usize::MAX,
+                ..Default::default()
+            },
+        )
+        .await;
+
+        let opts = OpenOptions::default();
+        let region = RegionImpl::open(REGION_NAME.to_string(), store_config, &opts)
+            .await
+            .unwrap()
+            .unwrap();
+
+        self.base = Some(FileTesterBase::with_region(region));
+    }
+}
+
+#[tokio::test]
+async fn test_truncate_basic() {
+    let dir = create_temp_dir("truncate-basic");
+    common_telemetry::init_default_ut_logging();
+    let store_dir = dir.path().to_str().unwrap();
+
+    let flush_switch = Arc::new(FlushSwitch::default());
+    let tester = TruncateTester::new(store_dir, flush_switch.clone()).await;
+
+    let data = [
+        (1000, Some("1000".to_string())),
+        (1001, Some("1001".to_string())),
+        (1002, Some("1002".to_string())),
+        (1003, Some("1003".to_string())),
+    ];
+
+    // Data in Memtable
+    let _ = tester.base().put(&data).await;
+    let res = tester.base().full_scan().await;
+    assert_eq!(4, res.len());
+
+    // Truncate region.
+    tester.truncate().await;
+
+    let res = tester.base().full_scan().await;
+    assert_eq!(0, res.len());
+}
+
+#[tokio::test]
+async fn test_put_data_after_truncate() {
+    let dir = create_temp_dir("put_data_after_truncate");
+    common_telemetry::init_default_ut_logging();
+    let store_dir = dir.path().to_str().unwrap();
+
+    let sst_dir = format!("{}/{}", store_dir, engine::region_sst_dir("", REGION_NAME));
+    let flush_switch = Arc::new(FlushSwitch::default());
+    let tester = TruncateTester::new(store_dir, flush_switch.clone()).await;
+
+    let data = [
+        (1000, Some("1000".to_string())),
+        (1001, Some("1001".to_string())),
+        (1002, None),
+        (1003, Some("1003".to_string())),
+    ];
+    let _ = tester.base().put(&data).await;
+
+    // Manually trigger flush.
+    tester.flush().await;
+    assert!(has_parquet_file(&sst_dir));
+
+    let data = [
+        (1002, Some("1002".to_string())),
+        (1004, Some("1004".to_string())),
+        (1005, Some("1005".to_string())),
+    ];
+    let _ = tester.base().put(&data).await;
+
+    // Truncate region.
+    tester.truncate().await;
+    let res = tester.base().full_scan().await;
+    assert_eq!(0, res.len());
+
+    let new_data = [
+        (1000, Some("0".to_string())),
+        (1001, Some("1".to_string())),
+        (1002, Some("2".to_string())),
+        (1003, Some("3".to_string())),
```

Comment:
I think we should use keys different from the previous put keys to avoid a new key override the old one.
```suggestion
        (1010, Some("0".to_string())),
        (1011, Some("1".to_string())),
        (1012, Some("2".to_string())),
        (1013, Some("3".to_string())),
```